### PR TITLE
OSL tweaks

### PR DESCRIPTION
**Pull request by luisbarrancos**

### CODE REVIEW

File: src/appleseed.shaders/src/appleseed/as_standard_surface.osl [hunk 1]
```diff
@@ -35,10 +35,7 @@ shader as_standard_surface
     string as_maya_classification = "drawdb/shader/surface:rendernode/appleseed/surface:shader/surface:swatch/AppleseedRenderSwatch",
     string help = "Standard Surface Shader",
     string icon = "asStandardSurface",
-    int as_maya_type_id = 0x001279d4,
-    string as_max_class_id = "651059681 943276521",
```
Comment (est77) [inline src/appleseed.shaders/src/appleseed/as_standard_surface.osl]: Do we want to remove these lines?

Comment (luisbarrancos) [reply src/appleseed.shaders/src/appleseed/as_standard_surface.osl]: No we don't, i'm not sure how that ended up there.

File: src/appleseed.shaders/src/appleseed/as_standard_surface.osl [hunk 2]
```diff
@@ -672,7 +686,7 @@ shader as_standard_surface
 
         if (in_fresnel_type == 0)
         {
-            if (compute_transmission)
+            if (!compute_coating && compute_transmission)
```
Comment (dictoon) [inline src/appleseed.shaders/src/appleseed/as_standard_surface.osl]: Isn't that a regression? @est77 removed this check that was causing things to become black when coating was applied, IIRC.

File: src/appleseed.shaders/src/appleseed/as_standard_surface.osl [hunk 3]
```diff
@@ -477,7 +491,7 @@ shader as_standard_surface
         string as_maya_attribute_name = "maximumRayDepth",
         string as_maya_attribute_short_name = "mrd",
         int min = 0,
-        int max = 100,
+        int max = 32,
```
Comment (dictoon) [inline src/appleseed.shaders/src/appleseed/as_standard_surface.osl]: Why this change? appleseed allows up to 100 bounces.

File: src/appleseed.shaders/src/appleseed/as_standard_surface.osl [hunk 4]
```diff
@@ -97,6 +97,20 @@ shader as_standard_surface
         int as_maya_attribute_hidden = 1,
         int gafferNoduleLayoutVisible = 0
     ]],
+    int in_sss_profile = 0
+    [[
+        string as_maya_attribute_name = "sssProfile",
+        string as_maya_attribute_short_name = "ssp",
+        string widget = "mapper",
+        string options = "Better Dipole:0|Directional Dipole:1|Gaussian:2|Normalized Diffusion:3|Standard Dipole:4|Random Walk:5",
```
Comment (dictoon) [inline src/appleseed.shaders/src/appleseed/as_standard_surface.osl]: In terms of profiles, in my opinion all plugins should only expose (in this order):

- "Gaussian"
- Better Dipole with the name "Dipole"
- "Normalized Diffusion"
- "Random Walk"

File: src/appleseed.shaders/src/appleseed/as_standard_surface.osl (hunk 2 above)
Comment (luisbarrancos) [reply src/appleseed.shaders/src/appleseed/as_standard_surface.osl]: When this shader was done, coating on glass would generate a huge amount of noise hence it was disabled since it was unusable. This was kind of suspended pending work on a mechanism to layer BxDFs, we then had as_layer() closure. I haven't revisited this particular case yet, but if @est77 removed this condition then i assume the coating on BTDF case is already correct.

Comment (est77) [reply src/appleseed.shaders/src/appleseed/as_standard_surface.osl]: We did some testing after removing the condition and results looked correct.
Do you have any particular scene where the result was incorrect?

File: src/appleseed.shaders/src/appleseed/as_blackbody.osl [hunk 5]
```diff
@@ -271,14 +271,14 @@ shader as_blackbody
...
-        out_color += in_specular_amount *
+        out_color += in_specular_weight *
             as_glossy(
                 "ggx",
                 Nn,
                 vector(0), // tangent vector
                 in_specular_roughness,
                 0.0, // specular spread
-                0.0, // anisotropy amount
+                0.0, // anisotropy weight
```
Comment (dictoon) [inline src/appleseed.shaders/src/appleseed/as_blackbody.osl]: That's an example of _amount_ being a more proper term than _weight_, in my opinion.

File: src/appleseed.shaders/src/appleseed/as_standard_surface.osl [hunk 6]
```diff
@@ -97,30 +97,44 @@ shader as_standard_surface
...
         int gafferNoduleLayoutVisible = 0
     ]],
-    int in_sss_maximum_ray_depth = 2
+    int in_subsurface_profile = 0
     [[
-        string as_maya_attribute_name = "sssMaximumRayDepth",
+        string as_maya_attribute_name = "subsurfaceProfile",
+        string as_maya_attribute_short_name = "ssp",
+        string widget = "mapper",
+        string options = "Gaussian:0|Dipole:1|Normalized Diffusion:2|:Random Walk:3",
```
Comment (dictoon) [inline src/appleseed.shaders/src/appleseed/as_standard_surface.osl]: There's an extra `:` before "Random Walk", is that OK?

File: src/appleseed.shaders/src/appleseed/as_standard_surface.osl [hunk 7]
```diff
@@ -788,15 +803,34 @@ shader as_standard_surface
...
+            }
+            else
+            {
+                subsurface_profile = "randomwalk";
+            }
+
             // Disable Fresnel weight since transmittance is computed earlier,
             // but ideally, the BSSRDF should be used.
-
+
```
Comment (dictoon) [inline src/appleseed.shaders/src/appleseed/as_standard_surface.osl]: Remove trailing spaces.

File: src/appleseed.shaders/src/appleseed/as_texture.osl [hunk 8]
```diff
@@ -424,69 +426,79 @@ shader as_texture
...
+                
 #ifdef DEBUG
-            string shadername = "";
-            getattribute("shader:shadername", shadername);
-            warning("[WARNING!]: Unknown OETF mode %d, in %s, %s:%d\n",
-                    in_eotf, shadername, __FILE__, __LINE__);
+                string shadername = "";
+                
+                getattribute("shader:shadername", shadername);
+                warning("[WARNING!]: Unknown OETF mode %d, in %s, %s:%d\n",
```
Comment (dictoon) [inline src/appleseed.shaders/src/appleseed/as_texture.osl]: Remove exclamation mark.
